src/base/matrix.rs

```diff
@@ -2357,3 +2357,27 @@ impl<T> super::alias::Matrix1<T> {
         scalar
     }
 }
+
+/// Provides methods for transforming a matrix into a vector with different algorithms
+impl<T, R: Dim, C: Dim, S: Storage<T, R, C>> Matrix<T, R, C, S> 
+where
+    T: Clone,
+{
+    /// Converts matrix into a vector by concatenating rows
+    pub fn into_vec(&self) -> Vec<T> {
+        let (num_rows, num_columns) = self.shape();
+        let mut resulted_vector = Vec::with_capacity(num_rows * num_columns);
+
+        for i in 0..num_rows {
+            for j in 0..num_columns {
+                unsafe {
+                    resulted_vector.push(self.get_unchecked((i, j)).clone());
+                }
```

Comment:
```suggestion
          resulted_vector.push(unsafe { self.get_unchecked((i, j)) }.clone());
```

Please explain why this is safe to do.

Reply:
As the method suggests (get_unchecked), there is no warranty that (i, j) are in bounds, that the operation does not fail or that the operation is memory safe. Now, the methods called by resulted_vector are not unsafe. If the resulted_vector is immutable, then it will result in compilation error, not that the operation is not memory safe. If the push exceeds vector capacity, then Rust is designed to reallocate space. I will modify.

Reply:
I observed that was the way transpose method access the elements of a matrix in matrix.rs so I wanted to maintain the format.

Reply:
Please add a comment in the code explaining why this is used and why it is safe.

Reply:
I've added two comments that should clarify the problem